ssl/d1_lib.c

```diff
@@ -246,16 +251,26 @@ void dtls1_start_timer(SSL *s)
     }
 #endif
 
-    /* If timer is not set, initialize duration with 1 second */
+    /* If timer is not set, initialize duration with 1 second or
+     * a user-specified value if the timer callback is installed. */
     if (s->d1->next_timeout.tv_sec == 0 && s->d1->next_timeout.tv_usec == 0) {
-        s->d1->timeout_duration = 1;
+
+        if (s->d1->timer_cb != NULL)
+            s->d1->timeout_duration_ms = s->d1->timer_cb(s, 1000);
```

Comment:
A possible overflow issue here. The type of timeout_duration_ms != return type of timer_cb.